src/main/scala/tech/cryptonomic/conseil/tezos/TezosDatabaseOperations.scala

```diff
@@ -47,11 +47,58 @@ object TezosDatabaseOperations extends LazyLogging {
     * @return         Future on database inserts.
     */
   def writeBlocks(blocks: List[Block]): DBIO[Unit] =
-      DBIO.seq(
-        Tables.Blocks          ++= blocks.map(RowConversion.convertBlock),
-        Tables.OperationGroups ++= blocks.flatMap(RowConversion.convertBlocksOperationGroups),
-        Tables.Operations      ++= blocks.flatMap(RowConversion.convertBlockOperations)
-      )
+    DBIO.seq(
+      Tables.Blocks          ++= blocks.map(RowConversion.convertBlock),
+      Tables.OperationGroups ++= blocks.flatMap(RowConversion.convertBlocksOperationGroups),
+      Tables.Operations      ++= blocks.flatMap(RowConversion.convertBlockOperations)
+    )
+
+  /**
+    * Writes association of account ids and block data to define accounts that needs update
+    * @param accountIds will have blocks, paired with correspoinding account ids to store
+    * @return Database action possibly returning the rows written (if available form the underlying driver)
+    */
+  def writeAccountsCheckpoint(accountIds: List[(BlockHash, Int, List[AccountId])]): DBIO[Option[Int]] =
+    Tables.AccountsCheckpoint ++= accountIds.flatMap((RowConversion.convertBlockAccountsAssociation _).tupled)
+
+  /**
+    * Removes  data on the accounts checkpoint table
+    * @return the database action to run
+    */
+  def cleanAccountsCheckpoint(): DBIO[Unit] = Tables.AccountsCheckpoint.schema.truncate
+
+  /**
+    * Reads the account ids in the checkpoint table, considering
+    * only those that at the latest block level (highest value)
+    * @return a database action that loads the list of relevant rows
+    */
+  def getLatestAccountsFromCheckpoint(implicit ec: ExecutionContext): DBIO[Map[BlockReference, List[AccountId]]] =
```

Comment:
I'm wondering whether we should put in some tolerance, i.e. don't fetch any account that was checkpointed more than, let's say, 1,000 blocks ago. This way, we might avoid a scenario where some accounts are just not fetchable and the list grows slowly with each iteration until it slows the whole process down. By adding a tolerance, we give up on an account if it simply cannot be fetched after a reasonable amount of time. It can be left to the DevOps personnel to check the checkpoints table for 'lost' accounts and perform some kind of manual intervention.

Reply:
I just reread the code so I can give you a more concrete example:
- Some account ID was picked up due to a regular expression match. However, the account ID is actually invalid and somehow trying to fetch it causes the Lorre process to crash.
- Lorre restarts and, due to checkpointing, it picks up the old invalid account ID as well, tries again to retrieve it and crashes again.
- This keeps happening and cripples the Lorre process.

For this edge case, my initial proposal was to give up on checkpointed accounts that were more than 'x' blocks old. 

However, after thinking about it, we should not bother about this edge case until our approach for fetching Tezos data settles down in the coming weeks. The contents of this PR are more than good enough for now.